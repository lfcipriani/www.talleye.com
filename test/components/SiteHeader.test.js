import { render, screen } from '@testing-library/react';
import SiteHeader from '../../components/SiteHeader';

beforeEach(() => {
  const useRouter = jest.spyOn(require('next/router'), 'useRouter');
  useRouter.mockImplementation(() => ({
    locale: 'en',
    defaultLocale: 'en',
  }));
});

test('renders website title from config', () => {
  render(<SiteHeader />);
  const title = screen.getByText(process.env.NEXT_PUBLIC_SITE_TITLE);
  expect(title).toBeInTheDocument();
});

test('renders the header menu', () => {
  render(<SiteHeader />);
  const menu = screen.getByText(/About/);
  expect(menu).toBeInTheDocument();
});
