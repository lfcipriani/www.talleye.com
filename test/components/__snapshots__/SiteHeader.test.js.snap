// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader snapshot renders Header 1`] = `
<header
  className="jsx-2912880461"
>
  <a
    className="jsx-2912880461 title"
    href="/"
    name="top"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span
      className="jsx-2912880461"
    >
      Luis Cipriani
    </span>
  </a>
  <nav
    className="jsx-2912880461"
  >
    <ul
      className="jsx-2912880461"
    >
      <li
        className="jsx-2912880461"
      >
        <a
          className="jsx-3248159136"
          href="/posts-rss.xml"
          onClick={[Function]}
          onMouseEnter={[Function]}
          target="_blank"
          title="RSS feed"
        >
          <span
            className="jsx-3248159136 visually-hidden"
          >
            RSS feed
          </span>
          <svg
            aria-hidden="true"
            className="jsx-3248159136 svgIcon"
            focusable="false"
            height="20"
            viewBox="0 0 256 256"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              className="jsx-3248159136"
              fill="none"
              height="256"
              width="256"
            />
            <path
              className="jsx-3248159136 stroke"
              d="M48,144a64,64,0,0,1,64,64"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
            />
            <path
              className="jsx-3248159136 stroke"
              d="M48,96A112,112,0,0,1,160,208"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
            />
            <path
              className="jsx-3248159136 stroke"
              d="M48,48A160,160,0,0,1,208,208"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
            />
            <circle
              className="jsx-3248159136 fill"
              cx="52"
              cy="204"
              r="12"
            />
          </svg>
        </a>
      </li>
      <li
        className="jsx-2912880461"
      >
        <a
          href="#"
          onClick={[Function]}
          title="Toggle to dark theme"
        >
          <span
            className="jsx-676870541 visually-hidden"
          >
            Toggle to dark theme
          </span>
          <svg
            aria-hidden="true"
            className="jsx-676870541 svgIcon"
            fill="#284B63"
            focusable="false"
            height="20"
            viewBox="0 0 256 256"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              className="jsx-676870541"
              fill="none"
              height="256"
              width="256"
            />
            <path
              className="jsx-676870541 stroke"
              d="M88,148a68,68,0,1,1,68,68H76a44,44,0,1,1,14.30583-85.62208"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
            />
            <path
              className="jsx-676870541 stroke"
              d="M46.99,138.92889A64.12919,64.12919,0,0,1,9.6218,94.37955l.00023-.001A64.0566,64.0566,0,0,0,86.37854,17.622l.00084-.00019A64.02478,64.02478,0,0,1,136,80q0,1.51431-.06961,3.01174"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
            />
          </svg>
        </a>
      </li>
      <li
        className="jsx-2912880461"
      >
        <a
          className="jsx-2912880461"
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
